### PR TITLE
Add API documentation for parser.parse(stream)

### PR DESCRIPTION
**Pull request by leonardfactory**
Adds a description for the `parser.parse(stream)` core method (#106). I thought that, given its relevance, it deserves a place on top of the list.

I added a simple example, with both success/failure cases and a tip about using `parser.desc` in order to have meaningful error messages.

English is not my main language, so any feedback/change request is highly appreciated

### CODE REVIEW

File: API.md [hunk 1]
```diff
@@ -262,6 +262,26 @@ parser.parse('accccc');
 
 These methods are all called off of existing parsers, not from the `Parsimmon` object itself. They all return new parsers, so you can chain as many of them together as you like.
 
+## parser.parse(stream)
+
+Apply the `parser` on the provided string `stream`, yielding an object that contains status and parsed result.
```
Comment (wavebeem) [inline API.md]: "the `parser`" => `parser`

"status" => "the status"

File: API.md [hunk 2]
```diff
@@ -262,6 +262,26 @@ parser.parse('accccc');
...
+
+If the parser succeeds, `status` is set to *true*, and the value will be available in the `value` property.
+
+If the parser fails, `status` will be *false*. Further information on the error can be found at `index` and `expected`. `index` represents the furthest reached offset; it has a 0-based character `offset` and 1-based `line` and `column` properties. `expected` lists all tried parsers that were available at the offset, but the stream couldn't continue with any of these.
+
+```javascript
+var parser =
+  Parsimmon.alt(
+    // Use `parser.desc(string)` in order to have meaningful failure messages
```
Comment (wavebeem) [inline API.md]: This is true, but it's a bit more complicated. Only the "low level" parsers should have descriptions or the error messages can actually get worse. I think it's probably fine to keep here though 😄

Comment (leonardfactory) [inline API.md]: Right, maybe we can leave the comment here and describe the `.desc` behaviour in depth in its own section?

Comment (wavebeem) [inline API.md]: Agreed. Opened https://github.com/jneen/parsimmon/issues/123 for that.

**Comment by wavebeem** [comment]
Thanks again!

**Pull request merged**